### ISSUE TITLE
Question: does the linter support compileOnly configurations?

### ISSUE THREAD

**Issue by krulls** [nebula-plugins/gradle-lint-plugin#46]
It seems like the rule 'unused-dependency' is not ranking a dependency with configuration "compileOnly" correctly.

I have a project with 
"compileOnly 'ojdbc-driver'" and of course I am using classes from that driver in my source. But I do not want other projects to receive that library transitive because it will be provided by the Oracle middleware. That is why I chose configuration "compileOnly" (see [here](https://gradle.org/blog/compile-only-dependencies/))

Running gradle-lint-plugin gives me the following warning:
"warning   unused-dependency                  one or more classes in com.oracle:ojdbc6:11.2.0.3.0 are required by your code directly"

If the linter does not support that configuration I hereby ask for implementation.

**Comment by krulls** [comment]
@jkschneider: Can you provide information?

**Comment by krulls** [comment]
I can at least confirm that version 0.30.X of the plugin is not able to accept ignorance like this
```
gradleLint.ignore
{
	compileOnly 'com.oracle:ojdbc6:' + versionOjdbc
}
```
Still getting warnings like "warning   unused-dependency                  one or more classes in com.oracle:ojdbc6:11.2.0.3.0 are required by your code directly"

**Comment by jkschneider** [comment]
@krulls You are awesome at discovering dependency problems. Thanks again!

**Comment by novaterata** [comment]
I just tried this plugin for the first time. We use dagger2, ANTLRv4 and lombok all very popular libraries that rely on compileOnly to not include the entire annotation processor or code generator in the released software. Running fixGradleLint  caused havok all over the place replacing all my compileOnly with runtime which broke everything. I eventually gave up and reverted all my changes. I don't consider this implemented.
